### PR TITLE
[SPARK-40510][PS] Implement `ddof` in `Series.cov`

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Implement `ddof` in `Series.cov`, by switch to `SF.covar`

### Why are the changes needed?
for API coverage

### Does this PR introduce _any_ user-facing change?
yes, `ddof` supported now

```
        >>> s1 = ps.Series([0.90010907, 0.13484424, 0.62036035])
        >>> s2 = ps.Series([0.12528585, 0.26962463, 0.51111198])
        >>> with ps.option_context("compute.ops_on_diff_frames", True):
        ...     s1.cov(s2)
        -0.016857...
        >>> with ps.option_context("compute.ops_on_diff_frames", True):
        ...     s1.cov(s2, ddof=2)
        -0.033715...
```

### How was this patch tested?
added UT

### CODE REVIEW

File: python/pyspark/pandas/series.py [hunk 1]
```diff
@@ -993,6 +993,11 @@ def cov(self, other: "Series", min_periods: Optional[int] = None) -> float:
             Series with which to compute the covariance.
         min_periods : int, optional
             Minimum number of observations needed to have a valid result.
+        ddof : int, default 1
+            Delta degrees of freedom.  The divisor used in calculations
```
Comment (itholic) [inline python/pyspark/pandas/series.py:997]: nit: there are two spaces between sentences.

File: python/pyspark/pandas/series.py [hunk 2]
```diff
@@ -1029,7 +1038,9 @@ def cov(self, other: "Series", min_periods: Optional[int] = None) -> float:
         if len(sdf.head(min_periods)) < min_periods:
             return np.nan
         else:
-            return sdf.select(F.covar_samp(*sdf.columns)).head(1)[0][0]
+            return sdf.select(SF.covar(F.col(sdf.columns[0]), F.col(sdf.columns[1]), ddof)).head(1)[
+                0
+            ][0]
```
Comment (itholic) [inline python/pyspark/pandas/series.py:1043]: Can we make it a bit more prettier?

e.g.

```python
            return sdf.select(
                SF.covar(F.col(sdf.columns[0]), F.col(sdf.columns[1]), ddof)).head(1)[0][0]
```

File: python/pyspark/pandas/series.py [hunk 3]
```diff
@@ -1002,19 +1007,23 @@ def cov(self, other: "Series", min_periods: Optional[int] = None) -> float:
...
+        -0.033715...
         """
         if not isinstance(other, Series):
             raise TypeError("unsupported type: %s" % type(other))
         if not np.issubdtype(self.dtype, np.number):  # type: ignore[arg-type]
             raise TypeError("unsupported dtype: %s" % self.dtype)
         if not np.issubdtype(other.dtype, np.number):  # type: ignore[arg-type]
             raise TypeError("unsupported dtype: %s" % other.dtype)
+        if not isinstance(ddof, int):
+            raise TypeError("ddof must be integer")
```
Comment (itholic) [inline python/pyspark/pandas/series.py:1026]: Maybe do we need to add a negative test case?

**Review by itholic — COMMENTED** [review]
Looks good otherwise

File: python/pyspark/pandas/series.py (hunk 2 above)
Comment (zhengruifeng) [reply python/pyspark/pandas/series.py:1043]: this was changed by `dev/reformat-python`...

File: python/pyspark/pandas/series.py (hunk 3 above)
Comment (zhengruifeng) [reply python/pyspark/pandas/series.py:1026]: nice, willl update soon

File: python/pyspark/pandas/series.py (hunk 1 above)
Comment (zhengruifeng) [reply python/pyspark/pandas/series.py:997]: good catch

File: python/pyspark/pandas/series.py (hunk 2 above)
Comment (itholic) [reply python/pyspark/pandas/series.py:1043]: Yeah, sometimes `black` reformats the code looks more ugly 😂

**Comment by zhengruifeng** [comment]
also cc @HyukjinKwon

**Review by HyukjinKwon — APPROVED** [review]

**Comment by zhengruifeng** [comment]
Merged into master, thank @HyukjinKwon @itholic for reivews!

**Pull request closed without merging**